### ISSUE TITLE
Upsert in DataStore

### ISSUE THREAD

**Issue by nubpro** [aws-amplify/amplify-js#5954]
It would be nice to have this feature - upsert. Update data if record exists, otherwise insert.

I think I'm as descriptive as I can be. This feature would eradicate lots of boilerplates in our production use!

**Comment by nubpro** [comment]
on second thought, we probably dont need this at all, since anyway, we had to query the item before we can update it.
